shared/src/main/scala/com/nitro/nmesos/commands/BaseCommand.scala

```diff
@@ -69,18 +69,12 @@ trait BaseCommand {
    * Create or update the Singularity request based on the diff between local and remote.
    */
   def updateSingularityRequestIfNeeded(remoteOpt: Option[SingularityRequest], local: SingularityRequest) = {
-
     log.debug("Comparing remote and local configuration...")
-
     remoteOpt match {
       case None =>
         log.info(s" No Mesos config found with id: '${local.id}'")
         manager.createSingularityRequest(local).map(_ => local)
 
-      case Some(remote) if (remote.instances != local.instances) =>
```

Comment:
There were edge case no considered here, moved the `scale` logic to ` scaleSingularityRequestIfNeeded`